vignettes/delay_distributions.Rmd

```diff
@@ -0,0 +1,155 @@
+
+For example, the density function for a Gamma distribution can be made generalisable to any set of values, by wrapping it within another function that accepts a vector of numbers.
+
+```{r}
+# wrap stats::dgamma() in a function
+# the Gamma distribution parameters are contained within dens_gamma()
+dens_gamma <- function(x) {
+  stats::dgamma(x = x, shape = 5, scale = 1)
+}
+
+# check over a vector of `x`
+dens_gamma(x = seq(10))
+```
+
+::: {.alert .alert-info}
+More information on working with functions, and especially anonymous functions, can be found in the [chapter on Functional Programming in _Advanced R_](https://adv-r.hadley.nz/fp.html).
+Users working with R > 4.1.0 can also use the new syntax for anonymous functions.
+:::
+
+## Passing delay distribution density to _cfr_ functions
+
+delay distribution density functions can be passed, as anonymous functions, to _cfr_ functions as shown with the example data provided with the package @camacho2014.
+Parameters for the onset-to-death distribution of Ebola virus disease are taken from @barry2018.
+
+```{r}
+# load package data
+data("ebola1976")
+
+# pass function wrapping dgamma to cfr_static()
+cfr_static(
+  data = ebola1976,
+  delay_density = function(x) dgamma(x, shape = 2.40, scale = 3.33)
+)
+```
+
+## Using other distribution representations
+
+While many R packages provide support for representing probability distributions, we focus on two examples, _distributional_ and _distcrete_, to show how closures wrapping these could be implemented, while wrapping the parameters from @barry2018.
+
+Users may wish to use these or similar packages for better management of distributions and parameters.
+See the [CRAN probability distributions task view](https://cran.r-project.org/view=Distributions) for more information on distribution packages suitable for different use cases.
+
+### Using _distributional_
+
+Note that the output of `density(<distribution>, x)` is a list containing a vector.
+
+```{r}
+# using {distributional} and parameters from Barry et al. 2018
+dist_onset_to_death_ebola <- dist_gamma(shape = 2.40, rate = 1.0 / 3.33)
+
+# wrap function and pass it to cfr_static()
+# unlist() required as density(<distribution>, x) is a list
+cfr_static(
+  data = ebola1976,
+  delay_density = function(x) unlist(density(dist_onset_to_death_ebola, x))
+)
+```
+
+### Using _distcrete_
+
+The _distcrete_ package provides support for discretised distributions.
+Here, we show an example for the discretised Gamma distribution.
+
+```{r}
+# using {distcrete} and parameters from Barry et al. 2018
+dist_onset_to_death_ebola <- distcrete(
+  name = "gamma", shape = 2.40, scale = 3.33, interval = 1
+)
+
+# pass density function to cfr_static()
+cfr_static(
+  data = ebola1976,
+  delay_density = function(x) dist_onset_to_death_ebola$d(x)
+)
```

Comment:
Thanks - I considered leaving it this way, but changed it for uniformity with the other examples. I can add a comment explaining why it can be passed without wrapping.

Reply:
Just looking into this - another reason the example is this way is that `delay_density` must be a function with a single argument, which the `distcrete_obj$d` function doesn't satisfy. I can change this check, but feel it would be better to constrain users in this way.

Reply:
My initial preference would be to allow both to be in line with what other common functions are doing (apply(), map(), etc.). Could you open an issue for this please so others can weigh in and we can take some time to consider both alternatives?

Reply:
Opened issue https://github.com/epiverse-trace/cfr/issues/89

Reply:
Fixed in https://github.com/epiverse-trace/cfr/pull/88/commits/08f870a666710c6500b1ba81075ddaee2003377a